clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -1601,6 +1601,34 @@ StringRef CIRGenModule::getMangledName(GlobalDecl GD) {
   return MangledDeclNames[CanonicalGD] = Result.first->first();
 }
 
+void CIRGenModule::buildTentativeDefinition(const VarDecl *D) {
+  assert(!D->getInit() && "Cannot emit definite definitions here!");
+
+  StringRef MangledName = getMangledName(D);
+  auto *GV = getGlobalValue(MangledName);
+
+  // TODO(cir): either remove or embelish this assert.
```

Comment:
What do you mean here?

Reply:
I'm basically not sure if we can have tentative definitions from something other than a global op. If this is the case, we eventually will have to remove this assertion. However, if this can only come from global ops, then It could be better written by first dynamically casting to a `cir::GlobalOp` and then asserting if the cast was successful.

I'll make this clearer.